documentation/leo/03_language.md

```diff
@@ -412,23 +412,127 @@ program test.aleo {
+If the key is not present, `0u64` is stored in `counter` and stored in `current_value`.
 
 #### set
+
 A set command, e.g. `Mapping::set(counter, addr, current_value + 1u64);`
 Sets the `addr` entry as `current_value + 1u64` in `counter`.
 
+# Leo Loops Documentation
+
+Leo supports the use of loops through `for` loops, which iterate over a range of values. Here's a detailed look at how
+loops work in Leo.
+
+## Basic Loops
+
+### The `for` Loop
+
+The `for` loop in Leo iterates over a range of values. This is similar to a `for` loop in many other programming
+languages, including Rust. Here is an example:
+
+```leo
+for i: u32 in 0u32..3u32 {
+    // loop body
+}
+```
+
+The loop starts at the first value (0 in this example) and ends at the last value exclusive (3 in this example, so the
+loop iterates over 0, 1, 2).
+
+In the loop body, you can use the loop variable `i`. In every iteration, `i` takes the value of the current iteration.
+In this example, `i` takes the values 0, 1, 2 in sequence.
+
+## Loop Behavior
+
+### Nested Loops
+
+You can nest loops within loops. For example:
+
+```leo
+for i: u32 in 0u32..2u32 {
+    for j: u32 in 0u32..2u32 {
+        // inner loop body
+    }
+}
+```
+
+In this case, for each value of `i`, the inner loop runs fully, iterating over its entire range of values.
+
+### Non-Literal Loop Bounds
+
+Leo does not currently support non-literal loop bounds. For example, the following code will fail:
+
+```leo
+for i:u64 in 0u64..amount {
+    // loop body
+}
+```
+
+The value of `amount` is not known at compile time, so the loop cannot be unrolled and compiled.
+
+### Return Statements in Loops
+
+Leo does not currently support return statements within loops. A function in Leo can only have one return statement, and
+it must be the last statement in the function. For example, the following code will fail:
+
+```leo
+for i: u32 in 0u32..9u32 {
+    return false;
+}
+```
+
+The `return` statement is inside the loop, which is not allowed in Leo.
+
+## Loop Errors
+
+### Loop Bounds
+
+Leo checks loop bounds at compile time. If the start bound is greater than the end bound, Leo will throw a compile
+error. For example, the following code will fail:
+
+```leo
+for i: u32 in 9u32..0u32 {
+    // loop body
+}
+```
+
+The start bound 9 is greater than the end bound 0, which is not valid.
+
+### Large Loop Bounds
+
+Leo does not support loop bounds larger than the maximum representable value of the loop variable's type. For example,
+the following code will fail:
+
+```leo
+for i:u64 in 0u64..1000000000000000000000000000000000000000000000000000000000000000000000000000000000000u64 {
+    // loop body
+}
+```
+
+The end bound is too large to be represented as a `u64`.
+
+## Notes
+
+1. Leo doesn't currently support the `break` and `continue` statements found in other languages such as
+   Rust. These statements allow for more complex control flow within loops, but are not currently
+   available in Leo.
+
```

Comment:
make these more specific: say exactly what the minimum and maximum loop requirements are or remove these docs